dlt/destinations/decorators.py

```diff
@@ -72,44 +112,113 @@ def destination(
+        spec (Type[CustomDestinationClientConfiguration]): Defines a configuration spec used
+            to inject arguments into the decorated function. Arguments not included in the spec will not be injected.
+
+        max_parallel_load_jobs (Optional[int]): Defines the maximum number of load jobs
+            that can run concurrently during the load process.
+
+        loader_parallelism_strategy (Optional[TLoaderParallelismStrategy]): Determines the load job parallelism strategy.
+            Can be "sequential" (equivalent to max_parallel_load_jobs=1), "table-sequential" (one load job per table at a time),
+            or "parallel".
+
+    Returns:
+        Callable[TDestinationCallableParams, _destination]: A callable that can be used to create a custom dlt destination instance.
+    """
+    ...
+
+
+@overload
+def destination(
+    destination_name: str,
+    /,
+    *,
+    destination_type: Optional[str] = None,
+    credentials: Optional[Any] = None,
+    **kwargs: Any,
+) -> AnyDestination:
+    """Instantiates a destination from the provided destination name and type by retrieving the corresponding
+    destination factory and initializing it with the given credentials and keyword arguments.
 
-        skip_dlt_columns_and_tables (bool): defines wether internal tables and columns will be fed into the custom destination function. This is set to True by default.
+    Args:
+        destination_name (str): The name of the destination instance to initialize.
 
-        max_table_nesting (int): defines how deep the normalizer will go to normalize nested fields on your data to create subtables. This overwrites any settings on your source and is set to zero to not create any nested tables by default.
+        destination_type (Optional[str]): The type of the destination to instantiate.
 
-        spec (Type[CustomDestinationClientConfiguration]): defines a configuration spec that will be used to to inject arguments into the decorated functions. Argument not in spec will not be injected
+        credentials (Optional[Any]): Credentials used to connect to the destination.
+            May be an instance of a credential class supported by the respective destination.
 
-        max_parallel_load_jobs (Optional[int]): how many load jobs at most will be running during the load
+        **kwargs (Any): Additional keyword arguments passed to the destination factory.
 
-        loader_parallelism_strategy (Optional[TLoaderParallelismStrategy]): Can be "sequential" which equals max_parallel_load_jobs=1, "table-sequential" where each table will have at most one loadjob at any given time and "parallel"
     Returns:
-        Any: A callable that can be used to create a dlt custom destination instance
+        AnyDestination: An initialized destination.
+    """
+    ...
+
+
+def destination(
+    func_or_name: Union[Optional[AnyFun], str] = None,
+    /,
+    loader_file_format: TLoaderFileFormat = None,
```

Comment:
let's enforce positional arguments: `*,` - it is right thing to do and skipping it does not help backward compat in any way